### PR TITLE
Serialize VRS features and add a workaround for vkd3d

### PR DESCRIPTION
**Pull request by hakzsam**
RADV enables a shader workaround due to a hardware bug based on these features.

### CODE REVIEW

File: cli/device.cpp [hunk 1]
```diff
@@ -368,20 +379,41 @@ bool VulkanDevice::init_device(const Options &opts)
...
+			requested_pdf2 = &replacement_pdf2;
+		}
+
+		if (application_info_promote_fragment_shading_rate(opts.application_info) &&
+		    find_pnext<VkPhysicalDeviceFragmentShadingRateFeaturesKHR>(
+				    VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_FRAGMENT_SHADING_RATE_FEATURES_KHR,
+				    opts.features->pNext) == nullptr)
+		{
+			replacement_fragment_shading_rate = { VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_FRAGMENT_SHADING_RATE_FEATURES_KHR };
+			replacement_pdf2 = *requested_pdf2;
```
Comment (HansKristian-Work) [inline cli/device.cpp:408]: The branch above copies PDF2 data from opts.features. Probably a good idea to harmonize the two.

File: cli/device.cpp [hunk 2]
```diff
@@ -368,20 +379,41 @@ bool VulkanDevice::init_device(const Options &opts)
...
+			replacement_pdf2.pNext = &replacement_fragment_shading_rate;
+			replacement_fragment_shading_rate.pNext = requested_pdf2->pNext;
```
Comment (HansKristian-Work) [inline cli/device.cpp:410]: If replacement_pdf2.pNext is written here. Wouldn't that potentially disturb requested_pdf2 as well? We probably need to write replacement_fragment_shading_rate->pNext, then the PDF2.

File: cli/device.cpp (hunk 1 above)
Comment (hakzsam) [reply cli/device.cpp:408]: Sounds good.

File: cli/device.cpp (hunk 2 above)
Comment (hakzsam) [reply cli/device.cpp:409]: Should be fixed now.

**Pull request merged**